### ISSUE TITLE
Per connection callback option?

### ISSUE THREAD

**Issue by BenAHammond** [tony-o/harmony#5]
I'd like the ability to run logic per-connection, prior to the connection being forwarded to the client.

Is this currently possible in this module, or would it need to be added?

My use case is pretty simple, I'd like to run this underneath IIS and manually patch in authenticated user information.

**Comment by tony-o** [comment]
you'd like to do `Client -> Harmony -> IIS` or are IIS and harmony switched around?

**Comment by BenAHammond** [comment]
Node iis hosted harmony server -> Linux hosted harmony clients

**Comment by tony-o** [comment]
Can you describe the environment a little further?  Are your users connecting directly to IIS and then to the node server or are they going through node to IIS ?  It would also help to know what you're trying to end goal is

**Comment by BenAHammond** [comment]
Sure thing.

The thought was to use https://github.com/tjanczuk/iisnode to host the
Harmony Server as a native process. With iis node, I could then configure
IIS for kerberos authentication and pass that response directly to the
client servers through Harmony.

Does Harmony work at the TCP or HTTP(s) layer?

**Comment by tony-o** [comment]
Harmony works at the network layer, let me think about how to do what you're describing.  A quick way to do that might be to just run a server to inject what you want into the HTTP request and then pass that connection directly to harmony running separately

**Comment by BenAHammond** [comment]
Yeah, that is the idea. I just don't know if the IIS kerberos handshake
will disrupt Harmony's request handling
